api/v1beta1/clusterclass_types.go

```diff
@@ -106,6 +106,10 @@ type ControlPlaneClass struct {
 	// +optional
 	MachineHealthCheck *MachineHealthCheckClass `json:"machineHealthCheck,omitempty"`
 
```

Comment:
Let's set all 3 namingStrategies in clusterclass-quick-start-runtimesdk.yaml for CAPD e2e tests to a non-default template.

GIves us some happy-path e2e test coverage for free

Reply:
I'm a bit on the fence. On one side it's easy to do on the other side it's already covered by unit tests and we usually try to avoid validating too much details in the e2e tests.

I think I wouldn't validate it. Basically just happy path in the sense of the test still works/ cluster is coming still up